src/main/java/org/uniovi/i3a/incimanager/rest/RESTController.java

```diff
@@ -41,23 +40,20 @@
 
 	@RequestMapping(value = "/sensor-feed", method = RequestMethod.POST, consumes = {
 			MediaType.APPLICATION_JSON_VALUE }, produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<String> processSensorRequest( @RequestBody Map<String, Object> payload ) {
+	public ResponseEntity<Object> processSensorRequest( @RequestBody Map<String, Object> payload ) {
 		
 		// Authentication of the agent that made the request.
-		val authenticationResponse = agentsConnection.executeQuery( new AgentsQueryFormatter( payload ).query() );
+		HttpResponse<JsonNode> authenticationResponse = 
+				agentsConnection
+				.executeQuery( new AgentsQueryFormatter( payload ).query() );
 		
 		if(authenticationResponse.getStatus() != HttpStatus.OK.value()) {
-			log.warn( "[ERROR] UNAUTHORIZED ACCESS: trying to access as: "
-						+ payload.get( "login" ) + " "
-						+ payload.get( "password" ) + " "
-						+ payload.get( "kind" ));
-			
-			return new ResponseEntity<String>("{\"response\":\"UNAUTHORIZED ACCESS WILL BE REPORTED\"}", HttpStatus.UNAUTHORIZED );
+			return new ResponseEntity<>( HttpStatus.UNAUTHORIZED );
 		}
 		
 		// Process the message in the request.
 		@SuppressWarnings({ "unchecked" })
-		val message = (LinkedHashMap<String, Object>) payload.get( "message" );
```

Comment:
Plase don't remove this val instance its from lombok project and helps inferring dynamic types.